### ISSUE TITLE
Feature request: Consider displaying posts with multiple images differently

### ISSUE THREAD

**Issue by markrprior** [Dimillian/IceCubesApp#1406]
I run a bot that now attaches 4 images to a post. Each of these images are formatted with an aspect ratio of 16:9, which seems to be the favoured mastodon aspect ratio. Most other iPhone apps (Mastodon, Toot!, Metatext, Mastoot, Ivory, Tusker, Trunks, Woolly and Mona) display these images in a 2x2 image grid where each element has the expected 16:9 aspect ratio. Icecubes also displays these 4 images in a 2x2 grid but crops the images to 1:1 which crops data for the graphs my bot is posting. This means the user has to open the images to see the data Icecubes has cropped. Mammoth also crops but interestingly only displays the first image and allows you to swipe across to the other 3 images (which is nice if it didn't crop). Tootle displays them as a 1x4 grid so crops them in such a way to make them totally useless without expanding them.

**Comment by Shanesan** [comment]
As far as real estate is concerned, a ![1:1 crop](https://user-images.githubusercontent.com/22644226/236239981-54b26f0b-1635-48f2-a22d-98de5bae0a67.jpeg) makes the best use of the room while making good aesthetic borders around the images. 16:9 use cases, like ![Mastodon's app](https://user-images.githubusercontent.com/22644226/236239996-35e1a94d-c3a8-4b23-983b-f8569ba9a0a1.jpeg), makes all the images smaller while still cropping the top and bottom losing some data in the preview ![vs the actual](https://user-images.githubusercontent.com/22644226/236240017-6a94d29e-40ff-4754-a5fc-5416dddd837c.jpeg).

Seeing the image in higher resolution, even if cropped, with better use of real estate as in Ice Cubes right now is my preferred method but only an opinion.

**Comment by Shanesan** [comment]
Hmmm. 

Does IceCubes support the [focal point](https://user-images.githubusercontent.com/22644226/236265285-ac27af87-89d8-4b17-be0c-a510957a7bbd.png) context clue from the Mastodon web app @Dimillian? 

Regardless, both apps are proven to crop the ratio to their own whims, but would focal point support help to maintain the aesthetic of the 1:1 ratio?

**Comment by markrprior** [comment]
@Shanesan IceCubes doesn't support setting a focal point.

When displaying them IceCubes crops portrait images across the middle, which is sort of better than mastodon (and metatext, mastoot, ivory, tusker, trunks, woolly, and mona) which does that but with its 16:9 ratio so you get even less. Toot! displays them better as they are portrait side by side but cropped on the edges to make 4 fit. Mammoth crops to square but displays them separately (you swipe to use the four) so slightly better than IceCubes as the cropped image is bigger. The only app I have installed that does 4 portrait orientation images without cropping is Tootle.

The mastodon web front end also crops but calckey's doesn't.

Example post so you can try it out for yourself is [https://ohai.social/@markrprior/110330803735963001](https://ohai.social/@markrprior/110330803735963001)

**Comment by Dimillian** [comment]
No focal point support for now. 
Also I want to rework it to make it a scrollable carrousel with support for portrait and landscape aspect ratio for images. So leaving this one but will probably be superseded with what I want to do.

**Comment by sh95014** [comment]
My complaint is that landscape images are all displayed in portrait crop:

![IMG_0529](https://github.com/Dimillian/IceCubesApp/assets/95387068/bb2eb201-9364-477a-b1d6-a587f1fde297)

which is almost disrespectful to the photographer/artist.

@Dimillian Do you want a separate issue filed for that, or would your rework cover this?

**Comment by Dimillian** [comment]
Yeah would be good to have the carousel displays them in landscape. But how do you mix both? I think Threads is favoring one or the other.

**Comment by sh95014** [comment]
Nicest and most general way is probably to pick a consistent scale ratio for both landscape and portrait images, then lay them out in a row. (Otherwise a landscape image scaled to the same height as a portrait image would take up a lot more screen real estate.)

Fitting the images into one of a few pre-defined "bento box layouts" is also possible, but I'd be wary of changing the ordering of the images...

**Comment by Dimillian** [comment]
Not sure it'll work. I just made the carrousel, before that it as a grid of 4 medias. I think the carrousel picking landscape if all or the majority of images are in landscape would be fine.
